File: cdk-from-cfn-testing/expected/condition_sub/python/ConditionSubConstruct.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_sns as sns
from constructs import Construct

"""
  Exercises Fn::Sub inside a Condition
"""
class ConditionSubConstruct(Construct):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id)

    # Conditions
    has_custom_suffix = f"""{props['envName']}-suffix""" == 'prod-suffix'

    # Resources
    topic = sns.CfnTopic(self, 'Topic',
        ) if has_custom_suffix else None
    if (topic is not None):

File: cdk-from-cfn-testing/expected/condition_sub/golang/ConditionSubConstruct.go
package main

import (
	"fmt"

	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	sns "github.com/aws/aws-cdk-go/awscdk/v2/awssns"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type ConditionSubConstructProps struct {
	EnvName *string
}

/// Exercises Fn::Sub inside a Condition
type ConditionSubConstruct struct {
	constructs.Construct
}

func NewConditionSubConstruct(scope constructs.Construct, id string, props *ConditionSubConstructProps) *ConditionSubConstruct {
	construct := constructs.NewConstruct(scope, &id)

	hasCustomSuffix := jsii.String(fmt.Sprintf("%v-suffix", props.EnvName)) == jsii.String("prod-suffix")

	sns.NewCfnTopic(
		construct,
		jsii.String("Topic"),
		&sns.CfnTopicProps{
		},
	)

	return &ConditionSubConstruct{
		Construct: construct,
	}
}

